slam-libraries/Makefile

```diff
@@ -60,13 +64,23 @@ appimage: buildorb
 	chmod 755 packaging/appimages/deploy/*.AppImage
 
 setupcarto:
+ifeq ("Darwin", "$(shell uname -s)")
+	cd viam-cartographer/scripts && ./setup_cartographer_macos.sh
+else
 	cd viam-cartographer/scripts && ./setup_cartographer.sh
+endif
+	
 
 buildcarto:
-	cd viam-cartographer/scripts && ./build_cartographer.sh && ./build_viam_cartographer.sh
+ifneq ($(wildcard viam-cartographer/build/.),)
+	cd viam-cartographer && ./scripts/rebuild_viam_cartographer.sh 
+else 
+	cd viam-cartographer && ./scripts/build_cartographer.sh && ./scripts/build_viam_cartographer.sh
```

Comment:
The difference between "build" and "rebuild" scripts here looks to be just two lines. Not a good pattern to just copy/paste that much other stuff when you could have the IF statement just be in the script itself.

Reply:
that makes sense, I can move the IF into the script